compose/src/commonMain/kotlin/com/powersync/compose/DatabaseState.kt

```diff
@@ -0,0 +1,22 @@
+package com.powersync.compose
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.State
+import androidx.compose.runtime.collectAsState
+import com.powersync.sync.SyncStatus
+import com.powersync.sync.SyncStatusData
+import kotlinx.coroutines.FlowPreview
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.debounce
+import kotlin.time.Duration
+
+@OptIn(FlowPreview::class)
+@Composable
+public fun SyncStatus.composeState(debounce: Duration=Duration.ZERO): State<SyncStatusData> {
+    var flow: Flow<SyncStatusData> = asFlow()
+    if (debounce.isPositive()) {
```

Comment:
If my understanding is correct: a subtle "quirk" with the `debounce` operator is no values are emitted until a period of inactivity equal to the time window is experienced. Depending on the window size, this could result in no updates being emitted during periods of high sync volume/updates. 

IIRC we use debounce in the demos, but maybe we should use the `throttle` helper if we include this in the compose module?

Reply:
I agree that we shouldn't use `debounce`, but `throttle` is internal and I don't think exposing it for this is worth it.

I've removed the parameter to debounce the status now. While we can always add it back later without breaking anyone, I wouldn't expect the status to update often enough to justify a debouncing logic.